background/background-worker.js

```diff
@@ -122,28 +126,39 @@ function getUsercssCompiler(preprocessor) {
         const pool = new Map();
         return Promise.resolve(doReplace(source));
 
-        function getValue(name, rgb) {
+        function getValue(name, rgbName) {
           if (!vars.hasOwnProperty(name)) {
             if (name.endsWith('-rgb')) {
-              return getValue(name.slice(0, -4), true);
+              return getValue(name.slice(0, -4), name);
             }
             return null;
           }
-          if (rgb) {
-            if (vars[name].type === 'color') {
-              const color = colorConverter.parse(vars[name].value);
-              if (!color) return null;
-              const {r, g, b} = color;
-              return `${r}, ${g}, ${b}`;
+          const {type, value} = vars[name];
+          switch (type) {
+            case 'color': {
+              let color = pool.get(rgbName || name);
+              if (color == null) {
+                color = colorConverter.parse(value);
+                if (color) {
+                  if (color.type === 'hsl') {
+                    color = colorConverter.HSVtoRGB(colorConverter.HSLtoHSV(color));
+                  }
+                  const {r, g, b} = color;
+                  color = rgbName
+                    ? `${r}, ${g}, ${b}`
+                    : `#${(0x1000000 + (r << 16) + (g << 8) + b).toString(16).slice(1)}`;
+                }
```

Comment:
I think we should move this logic into colorconverter.js:
https://github.com/openstyles/stylus/blob/7d93608186f6d694b911a6a16b9fc87228860581/vendor-overwrites/colorpicker/colorconverter.js#L19

So we can use it like:
```js
if (color) {
  color.a = null;
  color = converter.format(color, rgbName ? 'rgb-without-function' : 'hex');
}
```

Reply:
Yeah, sounds like something for someone to do it in the future. Meanwhile I'm thinking what to do about this PR as it solves a bug at the expense of the ability to set transparency via UI... Personally I think the use case is negligible because it was never an allowed feature in `uso` anyway.

Reply:
I'd still prefer to just follow the original format of the default value. If that color has transparency, enable it in the UI as well. It's not strictly compatible with USO but it's pretty much impossible for this to break any existing styles.

Reply:
Some progress: https://github.com/openstyles/stylus/pull/1200#issuecomment-790582832